dev/com.ibm.ws.microprofile.config.1.4/src/com/ibm/ws/microprofile/config14/impl/ConfigAccessorImpl.java

```diff
@@ -0,0 +1,222 @@
+//    }
+//
+//    /** {@inheritDoc} */
+//    @Override
+//    public ConfigAccessor<List<T>> asList() {
+//        ConfigAccessor<List<T>> accessor = new ConfigListAccessorImpl<>(propertyName, conversionType);
+//        return accessor;
+//    }
+//
+//    /** {@inheritDoc} */
+//    @Override
+//    public ConfigAccessor<Set<T>> asSet() {
+//        ConfigAccessor<Set<T>> accessor = new ConfigSetAccessorImpl<>(propertyName, conversionType);
+//        return accessor;
+//    }
+//
+//    /** {@inheritDoc} */
+//    @Override
+//    public ConfigAccessor<T> useConverter(Converter<T> converter) {
+//        this.converter = converter;
+//        return this;
+//    }
+//
+//    /** {@inheritDoc} */
+//    @Override
+//    public ConfigAccessor<T> withDefault(T value) {
+//        this.defaultValue = value;
+//        return this;
+//    }
+//
+//    /** {@inheritDoc} */
+//    @Override
+//    public ConfigAccessor<T> withStringDefault(String value) {
+//        this.defaultString = value;
+//        return this;
+//    }
+
+    /** {@inheritDoc} */
+    @Override
+    public ConfigAccessor<T> cacheFor(long value, TimeUnit timeUnit) {
+        this.cacheFor = value;
+        this.cacheForUnit = timeUnit;
+        return this;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public ConfigAccessor<T> evaluateVariables(boolean evaluateVariables) {
+        this.evaluateVariables = evaluateVariables;
+        return this;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public ConfigAccessor<T> addLookupSuffix(String suffixValue) {
+        this.suffixValue = suffixValue;
+        return this;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public ConfigAccessor<T> addLookupSuffix(ConfigAccessor<String> suffixAccessor) {
+        this.suffixAccessor = suffixAccessor;
+        return this;
+    }
+
+    public static String resolve(Config config, String raw) {
+        String resolved = raw;
+        //System.out.println("RAW: " + raw);
+        if (resolved != null) {
+            int evalStartIdx = resolved.indexOf(Config14Constants.EVAL_START_TOKEN);
+            while (evalStartIdx > -1) {
+                String part1 = resolved.substring(0, evalStartIdx);
+                String part2 = resolved.substring(evalStartIdx);
+                int evalEndIdx = part2.indexOf(Config14Constants.EVAL_END_TOKEN);
+                if (evalEndIdx < 0) {
+                    break; //can't find the end of the variable name so stop looking
+                }
+                String variableName = part2.substring(Config14Constants.EVAL_START_TOKEN.length(), evalEndIdx);
+                //System.out.println("VAR: " + variableName);
+                String value = config.getValue(variableName, String.class);
+                //System.out.println("VAL: " + value);
+
+                String part3 = part2.substring(evalEndIdx + Config14Constants.EVAL_END_TOKEN.length());
+                String part3resolved = resolve(config, part3);
```

Comment:
Recursive *and* iterative?

Reply:
you think i should make it more complicated?!